confuse.py

```diff
@@ -1526,6 +1548,8 @@ def as_template(value):
     elif isinstance(value, set):
         # convert to list to avoid hash related problems
         return Choice(list(value))
+    elif isinstance(value, type) and issubclass(value, type):
```

Comment:
Is this intended to check `issubclass(value, enum.Enum)` (if `SUPPORTS_ENUM`)? I think `issubclass(value, type)` is only true for metaclasses.

Reply:
You're right. Embarrassingly, this was not caught because my tests were also wrong... Fixed!